plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java

```diff
@@ -54,22 +55,38 @@ public CacheMediator createNode(IGraphicalEditPart part, AbstractMediator object
 			setElementToEdit(mediatorModel);
 			setCommonProperties(mediator, mediatorModel);
 
-			if (StringUtils.isNotBlank(mediator.getId())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_ID, mediator.getId());
-			}
-
-			if ("per-mediator".equals(mediator.getScope())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_SCOPE, CacheScope.PER_MEDIATOR);
-			} else if ("per-host".equals(mediator.getScope())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_SCOPE, CacheScope.PER_HOST);
-			}
 
 			if (mediator.isCollector()) {
 				executeSetValueCommand(CACHE_MEDIATOR__CACHE_ACTION, CacheAction.COLLECTOR);
 			} else {
+				String[] methods = mediator.getHTTPMethodsToCache();
+				if (methods.length != 0 && !methods[0].isEmpty()) {
+					StringBuilder method = new StringBuilder();
+					for (int i = 0; i < methods.length; i++) {
+						if (i != methods.length - 1) {
+							method.append(methods[i]).append(", ");
+						} else {
+							method.append(methods[i]);
+						}
+					}
+					executeSetValueCommand(CACHE_MEDIATOR__CACHE_PROTOCOL_METHODS, method.toString());
+				}
+				executeSetValueCommand(CACHE_MEDIATOR__CACHE_PROTOCOL_TYPE, mediator.getProtocolType());
+				executeSetValueCommand(CACHE_MEDIATOR__RESPONSE_CODES, mediator.getResponseCodes());
+				executeSetValueCommand(CACHE_MEDIATOR__MAX_MESSAGE_SIZE, mediator.getMaxMessageSize());
+
+				String[] headerstoExclude = mediator.getHeadersToExcludeInHash();
+				StringBuilder header = new StringBuilder();
+				for (int i = 0; i < headerstoExclude.length; i++) {
```

Comment:
Same as above. Do we need the if inside the for loop? We can simply loop until headerstoExclude.length and add the code in else after the loop.